source/fundamentals/serialization/poco.txt

```diff
@@ -207,22 +255,334 @@ Omit Empty Fields
+
+The {+driver-short+} includes the following ``IIdGenerator`` types:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 10
+
+   * - ``Id`` Field Data Type
+     - ``IIdGenerator`` Type
+   * - ``Guid`` value generated by the COMB algorithm
+     - ``CombGuidGenerator``
+   * - ``BsonObjectId``
+     - ``BsonObjectIdGenerator``
+
+In the following code example, if the ``Id`` property of the ``House`` class contains
+the default value (``null``), the driver uses the COMB algorithm to generate a
+unique value during serialization:
+
+.. code-block:: csharp
+   :copyable: false
+   :emphasize-lines: 3
+   
+   public class House
+   {
+       [BsonId(IdGenerator = typeof(CombGuidGenerator))]
+       public Guid Id { get; set; }
+   }
+
+.. note::
+   
+   In the previous code example, if the ``Id`` property didn't have the
+   ``[BsonId(IdGenerator)]`` attribute, the driver would generate a non-COMB GUID
+   to assign to the ``Id`` field.
+
+You can also specify an ``IIdGenerator`` type while registering the class map, as shown in
+the following example:
+
+.. code-block:: csharp
+   :copyable: true
+
+   BsonClassMap.RegisterClassMap<House>(classMap => 
+   {
+      classMap.AutoMap();
+      classMap.MapIdMember(h => h.Id).SetIdGenerator(CombGuidGenerator.Instance);
+   });
+
+.. tip:: Specify an ``IIdGenerator`` for Multiple Classes
+
+   You can use the ``RegisterIdGenerator()`` method to specify a single ``IIdGenerator``
+   for all ``Id`` properties of a certain data type. The following code example instructs
+   the driver to use the ``CombGuidGenerator`` type for all ``Guid`` IDs:
+
+   .. code-block:: csharp
+      :copyable: true
+   
+      BsonSerializer.RegisterIdGenerator(
+        typeof(Guid),
+        CombGuidGenerator.Instance
+      );
+
+The {+driver-short+} also includes ``IIdGenerator`` types that validate the ``Id``
+property and throw an exception if the ID is invalid. The following table lists these
+types:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 10 
+
+   * - ID Validation 
+     - ``IIdGenerator`` Type 
+   * - Not null 
+     - ``NullIdChecker``
+   * - Not all zeroes 
+     - ``ZeroIdChecker<T>``
+
+In the following code example, if the ``Id`` property of the ``House`` class contains
+the default value (``null``), the driver throws an exception:
+
+.. code-block:: csharp
+   :copyable: false
+   :emphasize-lines: 3
+   
+   public class House
+   {
+       [BsonId(IdGenerator = typeof(NullIdChecker))]
+       public Guid Id { get; set; }
+   }
+
+Customize DateTime Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize how the {+driver-short+} serializes ``DateTime`` properties, use the
+``[BsonDateTimeOptions()]`` attribute and specify the desired setting as an argument.
+
+If a ``DateTime`` property represents only a date, you can apply the
+``[BsonDateTimeOptions(DateOnly = true)]`` attribute to it.
+If you do, the driver won't perform any time-zone conversion on the value.
```

Comment:
S: time zone is two words - https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#t:~:text=as%20a%20noun.-,time%20zone,-Use%20time%20zone
```suggestion
If you do, the driver won't perform any time zone conversion on the value.
```